modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,249 @@
+				DeepLTranslatorConfiguration.class,
+				translatorPacket.getCompanyId());
+
+		String targetLanguageCode = _getLanguageCode(
+			translatorPacket.getTargetLanguageId());
+
+		List<String> supportedLanguages = _getSupportedLanguages(
+			deepLTranslatorConfiguration);
+
+		if (!_verifyLanguage(supportedLanguages, targetLanguageCode)) {
+			_log.error(
+				StringBundler.concat(
+					"No target language available for ", targetLanguageCode,
+					". Supported languages are: ",
+					StringUtil.merge(
+						supportedLanguages, StringPool.COMMA_AND_SPACE)));
+
+			return translatorPacket;
+		}
+
+		String sourceLanguageCode = _getLanguageCode(
+			translatorPacket.getSourceLanguageId());
+
+		Map<String, String> translatedFieldsMap = new HashMap<>();
+		Map<String, String> fieldsMap = translatorPacket.getFieldsMap();
+
+		for (Map.Entry<String, String> entry : fieldsMap.entrySet()) {
+			translatedFieldsMap.put(
+				entry.getKey(),
+				_translate(
+					deepLTranslatorConfiguration.url(),
+					deepLTranslatorConfiguration.authKey(), entry.getValue(),
+					sourceLanguageCode, targetLanguageCode));
+		}
+
+		return new TranslatorPacket() {
+
+			@Override
+			public long getCompanyId() {
+				return translatorPacket.getCompanyId();
+			}
+
+			@Override
+			public Map<String, String> getFieldsMap() {
+				return translatedFieldsMap;
+			}
+
+			@Override
+			public String getSourceLanguageId() {
+				return translatorPacket.getSourceLanguageId();
+			}
+
+			@Override
+			public String getTargetLanguageId() {
+				return translatorPacket.getTargetLanguageId();
+			}
+
+		};
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_deepLTranslatorConfiguration = ConfigurableUtil.createConfigurable(
+			DeepLTranslatorConfiguration.class, properties);
+	}
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		return StringUtil.toUpperCase(list.get(0));
+	}
+
+	private List<String> _getSupportedLanguages(
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration) {
+
+		try {
+			List<SupportedLanguage> supportedLanguages =
+				_deepLClient.getSupportedLanguages(
+					deepLTranslatorConfiguration.authKey(),
+					DeepLConstants.TARGET,
+					DeepLConstants.SUPPORTED_LANGUAGES_URL);
+			List<String> languages = new ArrayList<>();
+
+			supportedLanguages.forEach(
+				supportedLanguage -> languages.add(
+					supportedLanguage.getLanguage()));
+
+			return languages;
+		}
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return new ArrayList<>();
+		}
+		catch (TranslatorException translatorException) {
+			throw new RuntimeException(translatorException);
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
```

Comment:
When you have two catch blocks with the same body, you need to join them:
```
catch (PortalException | TranslatorException exception) {
    ...
}
```
Exceptions must be sorted alphabetically, and the variable name must be `exception`.

Reply:
Anyway, there's no need to catch these two exceptions here; you can annotate the method with `throws PortalException` and let it bubble up through the call chain.